### PR TITLE
Feat: Completed Auth Feature

### PR DESCRIPTION
**Pull request by vinc3nati**
- created auth context
- created helper login signup services
image
image

### CODE REVIEW

**Review by lomesshh — APPROVED** [review]
Code looks modular
Looks good to merge

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,21 @@
-import { Routes, Route } from "react-router-dom";
+import { Routes, Route, useLocation } from "react-router-dom";
 import Mockman from "mockman-js";
 import "./App.css";
 import { Home } from "./pages/Home/Home";
 import { Footer } from "./components/Footer/Footer";
+import { Auth } from "./pages/Auth/Auth";
```
Comment (anuj-kumary) [inline src/App.js:6]: Try to use barrel import. It will make the imports shorter

**Review by anuj-kumary — APPROVED** [review]
Your code is a very clean and good use of naming convention.
good to merge

File: src/pages/Auth/Auth.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React, { useEffect } from "react";
+import { useNavigate, useLocation } from "react-router-dom";
+import { Login } from "../../components/Auth/Login";
+import { Signup } from "../../components/Auth/Signup";
```
Comment (Shah-Arsalan) [inline src/pages/Auth/Auth.jsx:4]: Use barrel imports for efficiency

File: src/pages/Auth/Auth.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  }, [user.token]);
+
+  return (
+    <section id="auth">
+      <div className="auth-container">
+        <aside className="auth-aside">
+          <div className="auth-image-container">
+            <img
+              className="img img-responsive"
+              src="https://res.cloudinary.com/randomwave45/image/upload/v1648214717/notes_auth_e1mc51.png"
```
Comment (Shah-Arsalan) [inline src/pages/Auth/Auth.jsx:27]: Liked how you used cloudinary

File: src/pages/Auth/auth.css [hunk 4]
```diff
@@ -0,0 +1,68 @@
+#auth {
+  height: 100vh;
+  padding: 2rem 2.5rem;
+}
+
+.auth-container {
+  display: flex;
+  /* max-width: 850px; */
```
Comment (Shah-Arsalan) [inline src/pages/Auth/auth.css:8]: remove comments

**Review by Shah-Arsalan — APPROVED** [review]
Code looks really good  ! You can go ahead with the merge

File: src/App.js (hunk 1 above)
Comment (vinc3nati) [reply src/App.js:6]: sure

File: src/pages/Auth/Auth.jsx (hunk 2 above)
Comment (vinc3nati) [reply src/pages/Auth/Auth.jsx:4]: sure

**Pull request merged**